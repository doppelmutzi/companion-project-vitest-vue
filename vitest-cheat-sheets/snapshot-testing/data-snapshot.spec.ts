import { test, expect, vi } from "vitest";

test("snapshot testing", () => {
  const data = {
    id: 1,
    name: "John Doe",
    email: "john.doe@example.com",
  };

  expect(data).toMatchSnapshot();
});

// Vitest snapshot example with an object like above but also a mocked function
test("snapshot testing with a mocked function", () => {
  const person = {
    id: 1,
    name: "John Doe",
    email: "john.doe@example.com",
    contact: vi.fn(),
  };

  expect(person).toMatchSnapshot();
});
