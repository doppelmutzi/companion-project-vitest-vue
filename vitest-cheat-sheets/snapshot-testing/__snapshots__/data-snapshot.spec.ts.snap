// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot testing 1`] = `
{
  "email": "john.doe@example.com",
  "id": 1,
  "name": "John Doe",
}
`;

exports[`snapshot testing with a mocked function 1`] = `
{
  "contact": [MockFunction spy],
  "email": "john.doe@example.com",
  "id": 1,
  "name": "John Doe",
}
`;
